src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -49,15 +49,16 @@ private void Visit(CSharpSyntaxNode syntax, Binder enclosing)
         // Currently the types of these are restricted to only be whatever the syntax parameter is, plus any LocalFunctionStatementSyntax contained within it.
         // This may change if the language is extended to allow iterator lambdas, in which case the lambda would also be returned.
         // (lambdas currently throw a diagnostic in WithLambdaParametersBinder.GetIteratorElementType when a yield is used within them)
-        public static SmallDictionary<CSharpSyntaxNode, Binder> BuildMap(Symbol containingMemberOrLambda, CSharpSyntaxNode syntax, Binder enclosing, ArrayBuilder<CSharpSyntaxNode> methodsWithYields)
+        public static SmallDictionary<SyntaxNode, Binder> BuildMap(Symbol containingMemberOrLambda, SyntaxNode syntax, Binder enclosing, ArrayBuilder<SyntaxNode> methodsWithYields)
         {
             var builder = new LocalBinderFactory(containingMemberOrLambda, syntax, enclosing, methodsWithYields);
 
-            if (syntax is ExpressionSyntax)
+            var exprNode = syntax as ExpressionSyntax;
+            if (exprNode != null)
             {
                 var binder = new ExpressionVariableBinder(syntax, enclosing);
                 builder.AddToMap(syntax, binder);
-                builder.Visit(syntax, binder);
+                builder.Visit(exprNode, binder);
```

Comment:
Yes.  'Visit' Can only be called on a CSaarpSyntaxNode (the 'Visit/Accept' methods are not on SyntaxNode itself).  However, as we can see from this section, we only ever called Visit on a node that was known to be an ExpressionSyntax.  So, in this code, i would have had to cast (safely) the syntax node to a CSharpSyntaxNode. Instead of having an is+cast, i just replaced both with a single 'as' cast.